src/node_nanomsg.cc

```diff
@@ -191,35 +190,40 @@ NAN_METHOD(Err) {
   info.GetReturnValue().Set(Nan::New(nn_strerror(nn_errno())).ToLocalChecked());
 }
 
-typedef struct nanomsg_socket_s {
-  uv_poll_t poll_handle;
-  uv_os_sock_t sockfd;
-  Nan::Callback *callback;
-} nanomsg_socket_t;
+class NanoMsgPollCtx {
+  private:
+  public:
+    uv_poll_t poll_handle;
+    uv_os_sock_t sockfd;
+    Nan::Callback *callback;
+    NanoMsgPollCtx (int s, bool is_sender, Local<v8::Function> cb) {
+      callback = new Nan::Callback(cb);
+      size_t siz = sizeof(uv_os_sock_t);
+      nn_getsockopt(s, NN_SOL_SOCKET, is_sender ? NN_SNDFD : NN_RCVFD, &sockfd,
+          &siz);
+      poll_handle.data = this;
+    }
+    ~NanoMsgPollCtx () {
+      delete callback;
+    }
```

Comment:
I'd make the `callback` member an actual `Nan::Callback` instead of a pointer. Then you would not need this destructor, (which incidentally disables automatic generation of move constructors, which might or might not be desirable).